### ISSUE TITLE
Parser doesn't work if 3 or more hyphens in front matter

### ISSUE THREAD

**Issue by jamesrayman** [virchau13/tree-sitter-astro#25]
Here's an example in the form of a tree-sitter test:
```
====================
Three hyphens in frontmatter
====================

---
const myArray = [-1, -2, -3]
---

<h1>Hello world!</h1>

--------------------

(document
  (frontmatter
    (raw_text))
  (element
    (start_tag
      (tag_name))
    (text)
    (end_tag
      (tag_name))))

```
Here is the test output:
```
1 failure:

correct / - expected / + unexpected

  1. Three hyphens in frontmatter:

    (document
+     (ERROR
+       (raw_text)
+       (UNEXPECTED '3'))
-     (frontmatter
-       (raw_text))
      (element
        (start_tag
          (tag_name))
        (text)
        (end_tag
          (tag_name))))
```
If you make one of the numbers in `myArray` positive, the test passes. I'm pretty sure the bug sits somewhere in `scan_js_expr` in `src/scanner.c`.

**Comment by jamesrayman** [comment]
Feel free to open a pull request yourself. Otherwise, you'll have to wait for me or someone else to decide it's worth our time to fix. If you need a quick and dirty fix in the meantime, comments reset the hyphen count. For example, the following is highlighted correctly:
```
---
const myArray = [-1, -2, /* */ -3]
---
```

**Comment by virchau13** [comment]
Sorry for the wait. Fixed in ccd8826973bdd770136481e96554ee7c570f0fd4.
